### PR TITLE
Create FindAppointmentCommand

### LINKED ISSUE #102: Create FindAppointmentCommand
Add functionality to find appointments.

### PR DESCRIPTION
**Pull request by zaidansani**
Resolves #102, #105

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/FindAppointmentCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.FindAppointmentCommand;
+
+public class FindAppointmentCommandParserTest {
+
+    @Test
+    public void parse_noPrefix_failure() {
+        FindAppointmentCommandParser parser = new FindAppointmentCommandParser();;
```
Comment (Vangmay) [inline src/test/java/seedu/address/logic/parser/FindAppointmentCommandParserTest.java:15]: I see 2 semicolons in this line, I suggest please remove one.

File: src/test/java/seedu/address/logic/parser/FindAppointmentCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        assertParseFailure(parser, " appt Alex",
+            String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindAppointmentCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_unsupportedPrefix_failure() {
+        FindAppointmentCommandParser parser = new FindAppointmentCommandParser();;
```
Comment (Vangmay) [inline src/test/java/seedu/address/logic/parser/FindAppointmentCommandParserTest.java:22]: There are 2 semicolons in this line too.

File: src/main/java/seedu/address/logic/parser/FindAppointmentCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        ArgumentMultimap argMultimap =
+            ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                PREFIX_TAG, PREFIX_PERSON_ID, PREFIX_DATETIME,
+                PREFIX_APPOINTMENT_TYPE, PREFIX_SICKNESS, PREFIX_MEDICINE);
+
+        return new FindAppointmentCommand(parsePredicate(argMultimap));
+
+    }
+
+    private Predicate<Appointment> parsePredicate(ArgumentMultimap argMultimap) throws ParseException {
```
Comment (Vangmay) [inline src/main/java/seedu/address/logic/parser/FindAppointmentCommandParser.java:45]: Minor nit, but please add Javadoc comments here.

**Review by Vangmay — COMMENTED** [review]
LGTM, I have left a few comments that need to be fixed

File: src/test/java/seedu/address/logic/parser/FindAppointmentCommandParserTest.java (hunk 1 above)
Comment (zaidansani) [reply src/test/java/seedu/address/logic/parser/FindAppointmentCommandParserTest.java:15]: Surprising this passed checkstyle, thanks!

**Comment by Vangmay** [comment]
LGTM

**Pull request merged**